test/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java

```diff
@@ -177,6 +177,9 @@ private static String findAttachmentsAndValidate(final File file) {
       final int index = fileName.indexOf(classesRoot) + classesRoot.length();
       String className = fileName.substring(index);
       className = className.replace(File.separator, ".");
+      if( !className.endsWith(".class")) {
+        return "";
+      }
```

Comment:
I don't have the specific example anymore. It died when I ran it somehow. I think namely by running another build step first we can get some more configuration files or something. 

This check is just super brittle on general principle. Skipping over the non-class files is harmless. We still accomplish the goal of the test  by checking that the validators are done correctly. So we lose functionality and we gain some robustness so the test doesn't blow up needlessly. There is no reason for this test to enforce the restriction that only class files can be in the places it looks. That is an odd thing for a unit test to do, and is very constraining (and over-engineered I would say as building stuff now that we don't actually need or have a case for).

Lastly, this test is bad. If i recall correctly we don't assert anything and we print a lot of errors to system out. If you run the tests this is a good chunk of the output. Arguably this test needs a whole lot more work than this little band-aid.